tests/store/crud/test_read.py

```diff
@@ -272,3 +272,16 @@ def test_get_users_no_username(user_store: MockStore, user_email: str):
 
     # THEN the user should be returned
     assert users[0].email == user_email
+
+
+def test_get_latest_failed_job_for_analysis(job_store: MockStore):
```

Comment:
Would it be possible to test other behaviours like what happens when an analysis doesn't have a failed job? And potentially preferable to use a real Store, rather than a mock store.

Reply:
Added it :+1: